frontend/src/PortfolioInfo.jsx

```diff
@@ -0,0 +1,195 @@
+import { SidebarProvider, SidebarTrigger } from "./components/ui/Sidebar";
+import AppSidebar from "./components/AppSidebar";
+import SearchBar from "./components/SearchBar";
+import { UserInfo } from "./context/UserContext";
+import { useState, useEffect } from "react";
+import { BASE_URL } from "./lib/utils";
+import { useParams, useNavigate } from "react-router-dom";
+import { Pencil, X } from "lucide-react";
+
+const PortfolioInfo = () => {
+  const [portfolioData, setPortfolioData] = useState(null);
+  const [companyIds, setCompanyIds] = useState([]);
+  const [companiesData, setCompaniesData] = useState([]);
+  const [companiesStockData, setCompaniesStockData] = useState([]);
+  const [isEditingMode, setIsEditingMode] = useState(false);
+
+  const navigate = useNavigate();
+
+  const { fullName } = UserInfo();
+  const { id } = useParams();
+
+  const getPortfolioData = async () => {
+    const response = await fetch(`${BASE_URL}/portfolios/${id}`, {
+      method: "GET",
+      credentials: "include",
+    });
+    const data = await response.json();
+    setPortfolioData(data);
+    if (companyIds.length === 0) {
+      setCompanyIds(data.companiesIds);
+    }
+  };
+
+  const handleDelete = async (companyId) => {
+    await fetch(`${BASE_URL}/portfolios/${id}/${companyId}`, {
+      method: "DELETE",
+      credentials: "include",
+    });
+    getCompaniesData();
```

Comment:
It looks like your `useEffect()` hook is configured to rerun when `companyIds` changes, so this function gets called anyway. Is this duplicative work?